### ISSUE TITLE
Create a shared function which generates an addon URL

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-frontend#8563]
There are a number of places in our code where we generate an add-on URL using

```
`/addon/${addon.slug}/`
```

Some examples include `HeroRecommendation`,  `SearchResult`, `AddonReviewList`, `AddonSummaryCard`, and `AddonTitle`.

We should create a shared function (which should probably live in `src/amo/utils/index.js`) which accepts a `slug` and returns this `addonURL` and then make use of that function everywhere that we currently have this code.

**Comment by rbrishabh** [comment]
Hwy! Can I take this up?

**Comment by bobsilverberg** [comment]
Yes, @rbrishabh, you may. What you need to know should be included in the description above. Please note that a test should also be added for the new function that you will be creating as part of this patch. Thanks!

**Comment by rbrishabh** [comment]
Alright! Thank you!

**Comment by AlexandraMoga** [comment]
@bobsilverberg is there something specific that we should be testing here, or should we keep an eye on regressions across the site? 
Thanks

**Comment by bobsilverberg** [comment]
> @bobsilverberg is there something specific that we should be testing here, or should we keep an eye on regressions across the site?
> Thanks

Code was changed in a number of places, but it's all using the same new function. It's the code that generates a URL that points to an add-on. If you look at the list of changed files in #8571 you'll see which pages have changed. Are you able to do that, or would you like me to compile a list for you?

**Comment by AlexandraMoga** [comment]
@bobsilverberg I think you are referring to the affected files mentioned in https://github.com/mozilla/addons-frontend/pull/8571#issuecomment-530049562 . If that is the case, I  know what we are expected to test.

**Comment by bobsilverberg** [comment]
> @bobsilverberg I think you are referring to the affected files mentioned in [#8571 (comment)](https://github.com/mozilla/addons-frontend/pull/8571#issuecomment-530049562) . If that is the case, I know what we are expected to test.

Correct! Thanks @AlexandraMoga!

**Comment by AlexandraMoga** [comment]
I haven't noticed regressions in the pages that were likely to be affected by these change.
I'm marking the issue as verified fixed.
